example/main.dart

```diff
@@ -103,8 +100,11 @@ void main() async {
     test('throws custom exception', () async {
       final dioError = DioError(
         error: {'message': 'Some beautiful error!'},
-        response: Response(statusCode: 500),
-        type: DioErrorType.RESPONSE,
+        response: Response(
+          statusCode: 500,
+          request: RequestOptions(path: '/foo'),
```

Comment:
A bit of nitpicking but maybe passing constant declared at the beginning of the file - `path` will make more sense in this case just for the sake of consistency, though it essentially changes nothing.